File: src/com/colacelli/irclib/connection/listeners/OnChannelMessageListener.java
package com.colacelli.irclib.connection.listeners;

import com.colacelli.irclib.connection.Connection;
import com.colacelli.irclib.messages.ChannelMessage;

public interface OnChannelMessageListener extends Listener {
    Type TYPE = Type.CHANNEL_MESSAGE;

    @Override
    default Type getType() {
        return TYPE;
    }

    void onChannelMessage(Connection connection, ChannelMessage message);
}

File: src/com/colacelli/irclib/connection/listeners/OnChannelMessageListener.kt
package com.colacelli.irclib.connection.listeners

import com.colacelli.irclib.connection.Connection
import com.colacelli.irclib.messages.ChannelMessage

interface OnChannelMessageListener : Listener {
    fun onChannelMessage(connection: Connection, message: ChannelMessage)

    override val type: Listener.Type
        get() = TYPE

    companion object {
        val TYPE = Listener.Type.CHANNEL_MESSAGE
    }
}
